### PR TITLE
dep: update a whole bunch of things

### PR DESCRIPTION
**Pull request by ckipp01**
This pr updates the following:

  - various dependencies to the latest
  - adds in dependabot
  - updates java and checkout actions
  - updates mill to the latest 0.10.5
  - bumps scalajs to 1.x
  - corrects repo location in pomSettings

### CODE REVIEW

**Comment by ckipp01** [comment]
> @ckipp01 Also worth having https://github.com/scala-steward-org/scala-steward-action ?

We've actually started adding the com-lihaoyi repos to the public steward instance, but since the CI is also out of date for these, I think it's better to do one big pr like this, get the CI and everything up to date, and then once we do that we can enable steward for the repo. Then we won't get blasted with a bunch of prs.

**Comment by IdiosApps** [comment]
> > @ckipp01 Also worth having https://github.com/scala-steward-org/scala-steward-action ?
> 
> We've actually started adding the com-lihaoyi repos to the public steward instance, but since the CI is also out of date for these, I think it's better to do one big pr like this, get the CI and everything up to date, and then once we do that we can enable steward for the repo. Then we won't get blasted with a bunch of prs.

Oh, I didn't know there was a public instance. Reading about it now, e.g. your post [here](https://contributors.scala-lang.org/t/can-scala-maintain-a-public-scala-steward-instance/5754). I see what you mean about being blasted with PRs - the [public instance](https://github.com/VirtusLab/scala-steward-repos) has some state which Actions won't have.

Thanks!

**Review by lefou — APPROVED** [review]
Looks good to me. Thank you!

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "monthly"
```
Comment (lefou) [inline .github/dependabot.yml:7]:
```suggestion
      interval: "weekly"
```

**Pull request merged**